docs/configuration/external-databases.md

```diff
@@ -0,0 +1,138 @@
+# Using external databases
+
+
+You can use an external database for the cloud controller and uaa by providing following values:
+
+```yaml
+#@data/values
+---
+capi:
+  database:
+    adapter: postgres
+    host: <host>
+    port: <port>
+    user: <user>
+    password: <password>
+    name: <database>
+
+uaa:
+  database:
+    adapter: postgresql
+    host: <host>
+    port: <port>
+    user: <user>
+    password: <password>
+    name: <database>
+    ca_cert: <ca certificate for tls>
+```
+
+
+Please note to the **different spelling** of `postgres`  and `postgresql` for capi and uaa.
+
+
+## Limitations
+
+The traffic to an external database will not be encrypted. This will be changed in the near future.
+
+In case of uaa, tls can be enabled by providing `ca_cert`.
+
+## Setup
+
+As prerequisite, you need to execute the following steps to configure your postgres installation for cf-for-k8s:
+
+1. Set the environment variables
+
+    ```bash
+    export VALUES_FILE=<path to file from above>
+    export PGPASSWORD=<password of postgres super user>
+    export PGHOST=<host where postres is running>
+    ```
+
+2. Run the following script. It will
+   * create one database each for Cloud Controller and UAA
+   * create one user each for Cloud Controller and UAA
+   * activate the `citext` extension for each of these databases
+
+    ```bash
+    VALUES_JSON=$(yaml2json "$VALUES_FILE")
+    CCDB_USERNAME=$(jq -r '.capi.database.user' <<< "$VALUES_JSON")
+    CCDB_PASSWORD=$(jq -r '.capi.database.password' <<< "$VALUES_JSON")
+    CCDB_NAME=$(jq -r '.capi.database.name' <<< "$VALUES_JSON")
+    UAADB_USERNAME=$(jq -r '.uaa.database.user' <<< "$VALUES_JSON")
+    UAADB_PASSWORD=$(jq -r '.uaa.database.password' <<< "$VALUES_JSON")
+    UAADB_NAME=$(jq -r '.uaa.database.name' <<< "$VALUES_JSON")
+    cat > /tmp/setup_db.sql <<EOT
+    CREATE DATABASE ${CCDB_NAME};
+    CREATE ROLE ${CCDB_USERNAME} LOGIN PASSWORD '${CCDB_PASSWORD}';
+    CREATE DATABASE ${UAADB_NAME};
+    CREATE ROLE ${UAADB_USERNAME} LOGIN PASSWORD '${UAADB_PASSWORD}';
+    EOT
+    psql -U postgres -f /tmp/setup_db.sql
+    psql -U postgres -d "${CCDB_NAME}" -c "CREATE EXTENSION citext"
+    psql -U postgres -d "${UAADB_NAME}" -c "CREATE EXTENSION citext"
+    ```
+
+
+## Example
+
+In the following section, an external database is created using the bitnami postgresql helm chart. Please note that this setup is **not suitable for productive environments**.
+
+1. Install postgresql using helm
+
+    ```bash
+    helm repo add bitnami https://charts.bitnami.com/bitnami
+    helm install postgresql bitnami/postgresql
+    ```
+
+1. Create configuration file `db-values.yaml`
+
+    ```yaml
+    #@data/values
+    ---
+    capi:
+      database:
+        adapter: postgres
+        host: postgres-postgresql.default.svc.cluster.local
```

Comment:
Kubernetes could not find a service with this host. Following these instructions, the UAA and API server failed due to the incorrect service name.

We were able to get it to work by changing the host to the provided value from the Helm installation. The Helm chart installs a Postgres service with host of `postgresql.default.svc.cluster.local`.

Reply:
That's correct. I will change it.